tendrl/gluster_integration/objects/volume/atoms/delete/__init__.py

```diff
@@ -88,8 +88,25 @@ def run(self):
             try:
                 # Acquire lock before deleting the volume from etcd
                 # We are blocking till we acquire the lock
+                # the lock will live for 60 sec after which it will be released.
+                locked = False
                 lock = etcd.Lock(NS.etcd_orm.client, 'volume')
-                lock.acquire(blocking=True,lock_ttl=None)
+
+                while not locked:
+                    try:
+                        # with ttl set, lock will be blocked only for 60 sec
+                        # after which it will raise lock_expired exception.
+                        # if this is raised, we have to retry for lock
+                        lock.acquire(blocking=True,lock_ttl=60)
```

Comment:
We are releasing the lock as soon as we are done with the job. So the ttl of 60 sec doesn't come into picture in normal case. Only if there is abrupt stopping of process while the lock was acquired, this ttl kicks in and releases the lock.
So to be on the safer side have set ttl of 60 sec. Considering this will happen rarely, I feel it to be okay
@r0h4n Thoughts ?